Source/Core/IonResource.js

```diff
@@ -164,8 +164,13 @@ define([
         return result;
     };
 
-    IonResource.prototype.fetchImage = function (preferBlob, allowCrossOrigin) {
-        return Resource.prototype.fetchImage.call(this, this._isExternal ? preferBlob : true, allowCrossOrigin);
+    IonResource.prototype.fetchImage = function (options) {
+        if (!this._isExternal) {
+            options = defaultValue(options, {});
```

Comment:
Never use `defaultValue` when the second parameter will allocate an object.  `defaultValue` doesn't allow for short-circuiting so what you're really doing here is creating a new object and throwing it away.

That being said, we should never modify `options` objects passed in from the user anyway, so this function should always create a new options object and set values from the one passed in (if defined).

Reply:
That's surprising to me because it seems to common in our code. A search for `defaultValue\([\w]+, {}` shows 20 occurrences (such as in `EllipsoidTerrainProvider.js` and `ImageryLayer.js`). Should these be fixed too?

Reply:
Forgotten or missed, write up a new issue and we'll make it as "good first issue" so new contributors can get their feet wet.